hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerManager.java

```diff
@@ -107,7 +108,23 @@ public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerStatus(
     List<DatanodeDetails> filterDns = null;
     if (hosts.isPresent() && !hosts.get().isEmpty()) {
       filterDns = NodeUtils.mapHostnamesToDatanodes(nodeManager, hosts.get(),
-          useHostnames);
+          useHostnames).stream()
+          .filter(dn -> {
+            try {
+              NodeStatus nodeStatus = nodeManager.getNodeStatus(dn);
+              boolean isInService = nodeStatus.isInService();
+              boolean isHealthy = nodeStatus.isHealthy();
+              if (!isInService || !isHealthy) {
```

Comment:
```suggestion
            if (!nodeManager.getNodeStatus(dn) != NodeStatus.inServiceHealthy()) {
```

also the line 140-144 could be:
```java
 return nodeManager.getNodes(NodeStatus.inServiceHealthy()).stream()
	  .filter(dn -> shouldReturnDatanode(filterStatus, dn))
	  .map(dn -> getInfoProto((DatanodeInfo)dn, clientVersion))
	  .collect(Collectors.toList());
```

Makes the filter semantic be more consistent.